integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/vector/LuceneVectorFieldIT.java

```diff
@@ -0,0 +1,464 @@
+import static org.hibernate.search.util.impl.integrationtest.common.assertion.SearchResultAssert.assertThatQuery;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.hibernate.search.backend.lucene.LuceneExtension;
+import org.hibernate.search.engine.backend.document.DocumentElement;
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.IndexObjectFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaObjectField;
+import org.hibernate.search.engine.backend.types.ObjectStructure;
+import org.hibernate.search.engine.backend.types.Projectable;
+import org.hibernate.search.engine.backend.types.VectorSimilarity;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.extension.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import org.apache.lucene.document.Document;
+
+class LuceneVectorFieldIT {
+
+	private static final byte[] BYTE_VECTOR_1 = new byte[] { 1, 2, 3, 4 };
+	private static final byte[] BYTE_VECTOR_2 = new byte[] { 1, 1, 1, 1 };
+
+	private static final float[] FLOAT_VECTOR_1 = new float[] { 1.0f, 2.0f, 3.0f, 4.0f, 1.0f, 2.0f, 3.0f, 4.0f };
+	private static final float[] FLOAT_VECTOR_2 = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f };
+
+	@RegisterExtension
+	public final SearchSetupHelper setupHelper = SearchSetupHelper.create();
+
+	@Test
+	void simpleVectorSavedAndRetrieved() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimple( index );
+
+		SearchQuery<Document> query = index.createScope().query()
+				.select(
+						f -> f.extension( LuceneExtension.get() ).document()
+				)
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<Document> result = query.fetchAll().hits();
+		assertThat( result )
+				.hasSize( 2 )
+				.satisfies( containsDocument(
+						doc -> doc.hasField( "string", "keyword1" )
+								.hasVectorField( "byteVector", BYTE_VECTOR_1 )
+								.hasVectorField( "floatVector", FLOAT_VECTOR_1 )
+								.andOnlyInternalFields()
+				) )
+				.satisfies( containsDocument(
+						doc -> doc.hasField( "string", "keyword2" )
+								.hasVectorField( "byteVector", BYTE_VECTOR_2 )
+								.hasVectorField( "floatVector", FLOAT_VECTOR_2 )
+								.andOnlyInternalFields()
+				) );
+	}
+
+	@Test
+	void simpleVectorSavedAndRetrievedViaProjection() {
```

Comment:
Would be nice to rely on `FieldProjectionBaseIT` for that kind of things that are not specific to vector fields at all.